### DISCUSSION TITLE
"Unexpected character received waiting for next packet.  char=195  seq=1"

### DISCUSSION THREAD

**Discussion by sirquackiii** [TomNisbet/TommyPROM#30] [Q&A] [answered]
I'm using Ben Eater's EEPROM programmer with the AT28C56, and when I write to it, it goes to about 80% until it stops and gives out this error. I tried unlocking the EEPROM but nothing changed.

What am I doing wrong?

Thanks.

**Comment by TomNisbet** [comment, marked as answer]
If you are using Ben Eater's hardware with my software then it isn't going to work.  He uses different I/O pins and the two aren't compatible.  The TommyPROM repo has a special unlock sketch that will work with Ben's hardware, but otherwise the main TommyPROM sketch is based on my design.

The error message shows that it is failing on the first packet, so I'd suspect a problem with the configuration of the file transfer software.  I can help you sort that out if you wire up your hardware to match the TommyPROM specs as shown here: https://tomnisbet.github.io/TommyPROM/hardware/

**Reply by sirquackiii** [reply]
![image](https://user-images.githubusercontent.com/77940766/124365971-cbaaa100-dc00-11eb-87da-9f6449d67208.png)

When I press enter in the terminal, nothing happens, and I cannot type. I set it to the correct baud, so I'm not sure what I'm doing wrong.

**Reply by TomNisbet** [reply]
Did you change something?  It must have been working more than that before to get the error messages.  Other than setting the baud and the correct port, there isn't much I can think of that would prevent you from at least seeing a prompt when you reset the arduino.

**Reply by sirquackiii** [reply]
All I did was upload the unlock sketch to the arduino, nothing else.

**Reply by TomNisbet** [reply]
Maybe try loading a simple Hello World sketch that just initializes the serial port and writes a string to it.  Put the write in the loop section so that it has continuous output.  Also check to make sure you didn't connect anything to the D0 and D1 pins.  Remove the Arduino from the breadboard if it is a Nano.

**Reply by sirquackiii** [reply]
It's working.
![image](https://user-images.githubusercontent.com/77940766/124396742-4b4b7500-dcc0-11eb-9e89-e8ce9972bc19.png)

**Reply by TomNisbet** [reply]
I see that TeraTerm is set for two different serial ports in your screenshots.  Maybe that was the earlier problem.

**Reply by sirquackiii** [reply]
I uploaded the sketch and connected to the correct port, set it to the correct baud, and still nothing really shows up. Earlier it was displaying some garbage text.

![image](https://user-images.githubusercontent.com/77940766/124404184-ad22d380-dcee-11eb-9de7-5fcd9abaef45.png)

I also decided to open the serial monitor in the Arduino IDE and noticed that it worked there, but I couldn't just use that since there is no file transfer.

**Reply by TomNisbet** [reply]
I see that you started a new thread.  How did you resolve your communication problem?  It might help someone else.
